thrift/server.go

```diff
@@ -185,6 +186,17 @@ func (s *Server) handle(origCtx context.Context, handler handler, method string,
 		return err
 	}
 
+	// Buffer the response write so we can send a system error to the client.
+	// It's too late to send a system error by the time we write Arg3.
```

Comment:
Are we sure about this? I'm quite sure Muttley can send an error halfway through a response, and the clients see the error.